components/function-controller/design/extend_function_spec_proposal.md

```diff
@@ -0,0 +1,496 @@
+
+```
+### Option 4
+
+Option presented before with separated configurations (templates) for build and function.
+
+```yaml
+apiVersion: serverless.kyma-project.io/v1alpha2
+kind: Function
+metadata:
+  name: my-function
+  namespace: default
+  labels:
+    app.kubernetes.io/name: my-function
+spec:
+  runtime: nodejs16
+  source:
+    ...
+  resourceProfiles:
+    function: S / M / L / XL / ... | (empty)-> resources field has to be filled
+    build: S / M / L / XL / ... | (empty)-> resources field has to be filled
+  replicas: 1
+  scalingConfig:
+    min: 1
+    max: 2
+  templates:
+    functionPod:
+      metadata: # labels and annotations only for function pod
+        labels: 
+          app: my-app
+        annotations: 
+          fluentbit.io/parser: my-regex-parser
+          istio-injection: enabled
+      spec: 
+        resources: # optional... required if spec.resourceProfiles.function is empty
+          limits: ... #if_custom
+          requests: ... #if_custom
+        env: # function pod envs
+          - name: PASSWORD
+            valueFrom:
+              secretRef:
+                name: mysvc-passwords
+                key: password
+          - name: EXTERNAL_API_URL
+            valueFrom:
+              configmapRef:
+                name: mysvc-configuration
+                key: URL
+        volumeMounts: ...
+          - name: config
+            mountPath: /etc/config/
+          - name: search-index
+            mountPath: /etc/index
+      volumes:
+        - name: search-index
+            nfs:
+              path: /path-to-index
+              readOnly: true
+              server: localhost
+        - name: config
+          configmap:
+            name: function-configuration
+            items:
+              - key: config
+                path: config.yaml
+    buildPod:
+      metadata: ...# labels and annotations only for build pod
+      spec: 
+        resources: # optional... required if spec.resourceProfiles.build is empty
+          limits: ... #if_custom
+          requests: ... #if_custom
+        env: ...
+        volumeMounts: ...
+      volumes: ...
+```
+
+
+### Compromise
+
+ - clearly separate configuration of the build stage in the spec.
+ - add simple, [purpose focused](https://blogs.sap.com/2022/07/12/the-new-way-to-consume-service-bindings-on-kyma-runtime/) `serviceBindings` 
+ - volume mounts separated to a different place, for more advanced case (implemented once requested) 
+ - don't group labels and annotations under metadata. 
+
+```yaml
+apiVersion: serverless.kyma-project.io/v1alpha2
+kind: Function
+metadata:
+  name: my-function
+  namespace: default
+  labels:
+    app.kubernetes.io/name: my-function
+spec:
+  runtime: nodejs16
+  source:
+    ...
+  replicas: 1
+  scalingConfig:
+    min: 1
+    max: 2
+
+  labels: #?
+    app: my-app
+  annotations: #?
+    fluentbit.io/parser: my-regex-parser
+    istio-injection: enabled
+
+  serviceBindings: 
```

Comment:
I'm wondering if that explicit relation to the ServiceBinding concept is good. That will introduce a dependency to it and you would need to explicitly test that (have an app using the specific libraries to interpret the metadata of a binding, to assure that actual serviceBindings can be consumed)
In contrast, if someone just wants to consume a 3party API using secret mounts he will be confused and wonder if that mechanism can be used for that.
So what are the actual parts which are specific to ServiceBindings? As far as I can see the defaulting of the mountPath to the SERVICE_BINDING_ROOT only, right?
We could either drop it fully and except a little loss in convenience

```yaml
secretMounts
  - secretName: icke
    mountPath: "myMountPath"
```
or we could add it as a capability to the binding itself?

```yaml
secretMounts
  - secretName: icke
     isServiceBinding: true
```

or more generic but coupled:
```yaml
secretMounts
  - fromSecret:
       name: icke
     mountPath: "/icke"
  - fromServiceBinding:
       name: "icke"
```

Reply:
Thank you @a-thaler 
In order to avoid implicit association with service bindings (https://github.com/SAP/sap-btp-service-operator#service-binding or https://servicebinding.io/) and user confusion I think the best option would be to simply enable the secret mount.

If user wants to use specialised libraries we would need to extra set the env variable, for example:

```yaml
secretMounts
  - secretName: icke
    mountPath: "/foo" #required.. no assumptions/validations towards SERVICE_BINDING_ROOT env value
    
 envs:
  - name: SERVICE_BINDING_ROOT
     value: "/foo"  #set explicitely if user wants to use a specialised library that expects it      
```

Reply:
the fact that `mountPath` and SERVICE_BINDING_ROOT value match indicates that this secret mount will be consumed in an "Service binding"- way via specialised library. 
But this is soft indication, not done by our API.
I think its sufficiently convenient